lib/galaxy/jobs/runners/__init__.py

```diff
@@ -728,6 +728,8 @@ def finish_job(self, job_state):
         """
         galaxy_id_tag = job_state.job_wrapper.get_id_tag()
         external_job_id = job_state.job_id
+        log.debug("(%s/%s) Job execution complete, performing job finish tasks", galaxy_id_tag, external_job_id)
+        job_state.job_wrapper.change_state(model.Job.states.FINISHING)
```

Comment:
Ugh - this line is going to be a big performance hit for relatively little gain from the user perspective 😟.